### ISSUE TITLE
Add admin user is broken

### ISSUE THREAD

**Issue by dadiorchen** [Greenstand/treetracker-admin-api#432]
Can not add a new admin panel user. 
![image (3)](https://user-images.githubusercontent.com/5744708/95936049-3adc2d00-0e07-11eb-9d8e-8544bd28fc54.png)
How to reproduce:
login: http://test.admin.treetracker.org account: admin:admin

**Comment by dadiorchen** [comment]
@deepwinter looks like this might be related to the database permission settings? On the test env, I found this error.
```
Oct 17 02:40:43 test-core-services-2 node[588]: testtest
Oct 17 02:40:43 test-core-services-2 node[588]: insert: insert into admin_user (user_name,first_name,last_name,email,passwor
d_hash,active) values ('test8','test','test','test@mail.com','undefined','true')
Oct 17 02:40:43 test-core-services-2 node[588]: { error: permission denied for table admin_user_role
Oct 17 02:40:43 test-core-services-2 node[588]:     at Connection.parseE (/root/treetracker-admin-api/node_modules/pg/lib/co
nnection.js:604:11)
Oct 17 02:40:43 test-core-services-2 node[588]:     at Connection.parseMessage (/root/treetracker-admin-api/node_modules/pg/
lib/connection.js:401:19)
Oct 17 02:40:43 test-core-services-2 node[588]:     at TLSSocket.<anonymous> (/root/treetracker-admin-api/node_modules/pg/li
b/connection.js:121:22)
Oct 17 02:40:43 test-core-services-2 node[588]:     at emitOne (events.js:116:13)
Oct 17 02:40:43 test-core-services-2 node[588]:     at TLSSocket.emit (events.js:211:7)
Oct 17 02:40:43 test-core-services-2 node[588]:     at addChunk (_stream_readable.js:263:12)
Oct 17 02:40:43 test-core-services-2 node[588]:     at readableAddChunk (_stream_readable.js:250:11)
Oct 17 02:40:43 test-core-services-2 node[588]:     at TLSSocket.Readable.push (_stream_readable.js:208:10)
Oct 17 02:40:43 test-core-services-2 node[588]:     at TLSWrap.onread (net.js:601:20)
Oct 17 02:40:43 test-core-services-2 node[588]:   name: 'error',
Oct 17 02:40:43 test-core-services-2 node[588]:   length: 103,
Oct 17 02:40:43 test-core-services-2 node[588]:   severity: 'ERROR',
Oct 17 02:40:43 test-core-services-2 node[588]:   code: '42501',
Oct 17 02:40:43 test-core-services-2 node[588]:   detail: undefined,
Oct 17 02:40:43 test-core-services-2 node[588]:   hint: undefined,
Oct 17 02:40:43 test-core-services-2 node[588]:   position: undefined,
Oct 17 02:40:43 test-core-services-2 node[588]:   internalPosition: undefined,
Oct 17 02:40:43 test-core-services-2 node[588]:   internalQuery: undefined,
Oct 17 02:40:43 test-core-services-2 node[588]:   where: undefined,
Oct 17 02:40:43 test-core-services-2 node[588]:   schema: undefined,
Oct 17 02:40:43 test-core-services-2 node[588]:   table: undefined,
Oct 17 02:40:43 test-core-services-2 node[588]:   column: undefined,
Oct 17 02:40:43 test-core-services-2 node[588]:   dataType: undefined,
Oct 17 02:40:43 test-core-services-2 node[588]:   constraint: undefined,
Oct 17 02:40:43 test-core-services-2 node[588]:   file: 'aclchk.c',
Oct 17 02:40:43 test-core-services-2 node[588]:   line: '3483',
Oct 17 02:40:43 test-core-services-2 node[588]:   routine: 'aclcheck_error' }
```

**Comment by dadiorchen** [comment]
And there is another thing we need to double-check. The `active` column in `user_admin` table, the case that we have created a user account and generated a password for them, but we can not still log in with it, that might because we messed up the `active` column to accidentally set it to false, so the user can not log in.

**Comment by ZavenArra** [comment]
what's the query that is being executed on admin_user_role?  treetracker has insert, select, and update on this table.

**Comment by dadiorchen** [comment]
Got it, there is a delete operation. 
https://github.com/Greenstand/treetracker-admin/blob/896ed35b06e32844f66032d8a48b2134a232e3d9/server/src/js/auth.js#L339-L345
I can delete the `delete` sentence.
Which branch did we deploy to the production env? `release`? I can consider creating a fix on the branch we are releasing.

**Comment by nmcharlton** [comment]
@dadiorchen Are you working on this?

In order to avoid `delete` actions on the db, we need to make proper use of the `active` column in this (and other) tables, and deal with duplicate entries in a cleaner way.

We should set `active` to `true` when the role is assigned to the user and `false` when it is removed from the user.
We could also consider adding a `UNIQUE` constraint to the `role_id`/`admin_user_id` pair in the db to prevent duplicates.

@deepwinter Do you agree with this approach?

**Comment by dadiorchen** [comment]
@nmcharlton great you can take this, just a suggestion, we'd better fix this against the release branch so we can launch this bug fix rather than the whole features developed on the master branch, right?

**Comment by nmcharlton** [comment]
Merged into `master` under #452
